src/Parser/Core/Modules/DamageTaken.js

```diff
@@ -20,12 +20,20 @@ class DamageTaken extends Module {
   get total() {
     return this._total;
   }
+
   _byAbility = {};
   byAbility(spellId) {
+    if(!this._byAbility[spellId]) {
+      this._byAbility[spellId] = new DamageValue(0, 0, 0);
```

Comment:
I would prefer returning `new DamageValue(0, 0, 0);` making this simpler, storing it permanently in the array seems to have no benefits.

Reply:
I went back and forth between that way and the way you see it implemented. I ended up going with this way because I rationalized this would be better in the case of multiple gets for the same spell that did 0 healing. In practice the benefit of this is probably indistinguishable from zero. 

I'll change it to your way, you're right it's simpler.